test/unittest/base/test_jsonization.py

```diff
@@ -0,0 +1,70 @@
+# import superduperdb as s
+
+import typing as t
+
+import pydantic as pd
+import pytest
+
+IS_2 = pd.__version__.startswith('2')
+
+
+@pytest.mark.skipif(not IS_2, reason='test requires pydantic 2')
+def test_jsonization1():
+    class Blob(pd.BaseModel):
+        uri: str = ''
+        type_id: t.Literal['blob1'] = 'blob1'
+        contents: bytes = bytes()
+
+        @pd.model_serializer
+        def ser_model(self) -> t.Dict[str, t.Any]:
+            return {'uri': self.uri, 'type_id': self.type_id}
+
+    blob = Blob(url='s')
+    s = blob.dict()
+    assert s == {'uri': '', 'type_id': 'blob1'}
+
+
+@pytest.mark.skipif(not IS_2, reason='test requires pydantic 2')
+def test_jsonization2():
+    class Blob:
+        def __init__(self, contents):
+            self.contents = contents
+
+    class Document(pd.BaseModel):
+        uri: str = ''
+        type_id: t.Literal['doc'] = 'doc'
+        blob: Blob = Blob(None)
+
+        class Config:
+            arbitrary_types_allowed = True
+
+        @pd.model_serializer
+        def ser_model(self) -> t.Dict[str, t.Any]:
+            return {'uri': self.uri, 'type_id': self.type_id}
+
+    doc = Document()
+    s = doc.dict()
+    assert s == {'uri': '', 'type_id': 'doc'}
+
+
+@pytest.mark.skipif(not IS_2, reason='test requires pydantic 2')
+def test_jsonization3():
```

Comment:
Is there a difference between '3' and '2'? (Sorry, I can't see it, maybe more coffee needed...)

Reply:
Oh,  probably not, I pulled this from a branch in development than I am abandoning.  I'll remove that, sorry!